### ISSUE TITLE
Make reselect work for null values

### ISSUE THREAD

**Issue by denisidoro** [beyondeye/Reduks#12]
`withField` requires a non-null value:
```fun<I1 : Any> withField(fn: S.() -> I1) = SelectorForP2<S, I0, I1>(si0, InputField(fn))```, where `I1: Any`

Say we have a state like:
```data class MyState(val apiResult: ApiResult?)```

Writing ```SelectorBuilder<MyState>().withField { apiResult }``` won't compile because `ApiResult? is not a subtype of Any`

Is there another way to circumvent this?

**Comment by beyondeye** [comment]
I will look into it. Personally I prefer to avoid nullable field in the state

**Comment by beyondeye** [comment]
as a work around you can define 
```kotlin
class Opt<T>(val it:T?)
data class MyState(val apiResult:Opt<ApiResult>)
```

**Comment by beyondeye** [comment]
I did a quick check.  The  problem is onChangeIn() that will not work anymore. Also the logger middleware has some problems with nulls.
TLDR: if you really need nulls in the state, wrap it with an Opt class

**Comment by denisidoro** [comment]
Fair enough. Thanks!
